core/src/main/java/com/amazon/opendistroforelasticsearch/sql/expression/operator/predicate/BinaryPredicateOperator.java

```diff
@@ -475,4 +513,34 @@ public String toString() {
       }
     };
   }
+
+  private static <T, R> FunctionBuilder likePattern(FunctionName functionName,
+                                                    BiFunction<T, T, R> function,
+                                                    Function<ExprValue, T> observer,
+                                                    ExprType returnType) {
```

Comment:
Just did a quick look and found this method is very similar to others, ex. `compareValue`, `equalPredicate` and `binaryPredicate`. The `equalPredicate` method signature looks most generic. I'm curious is there any chance to only keep or reuse a most generic function generator method.

Reply:
Yes that's also I'm considering, I will take a look into it

Reply:
Used generic function bundles generating method `binaryOperator` in `OperatorUtils`, but some for special cases are remained and added java doc for them.